ui/src/components/header/HeaderMenu.js

```diff
@@ -87,68 +92,103 @@ const HeaderMenuUserDiv = styled.div`
-    });
-    if (props.pending) {
-        if (Object.keys(props.pending).length !== 0) {
-            icon = 'notification-solid';
-        }
+const ReviewNotificationSpan = styled.span`
+    position: absolute;
+    top: 10px;
+    right: 90px;
+    width: 15px;
+    height: 15px;
+    background-color: red;
+    border-radius: 50px;
+`;
+
+class HeaderMenu extends React.Component {
+    constructor(props) {
+        super(props);
+        this.api = API();
+        this.state = {
+            roleGroupReviewFeatureFlag: false,
+        };
     }
-    let menuItems =
-        props.headerDetails &&
-        props.headerDetails.headerLinks &&
-        props.headerDetails.headerLinks.map((headerLink, idx) => {
-            return (
-                <MenuItemDiv key={idx}>
-                    <StyledAnchorDiv
-                        data-testid='menu'
-                        onClick={() =>
-                            window.open(headerLink.url, headerLink.target)
-                        }
-                    >
-                        {headerLink.title}
-                    </StyledAnchorDiv>
-                </MenuItemDiv>
-            );
+
+    onClickFunction(route) {
+        this.props.router.push(route, route);
+    }
+
+    componentDidMount() {
+        this.props.getUserPendingMembers();
+        this.props.getReviewRoles();
+        this.props.getReviewGroups();
+        this.api.getPageFeatureFlag('roleGroupReview').then((data) => {
+            this.setState({
+                roleGroupReviewFeatureFlag: data['roleGroupReviewFeatureFlag'],
+            });
         });
-    return (
-        <HeaderMenuDiv data-testid='header-menu'>
-            <Icon
-                icon={icon}
-                isLink
-                onClick={() => router.push(PageUtils.workflowAdminPage())}
-                size={'25px'}
-                color={colors.white}
-            />
-
-            <HeaderMenuUserDiv>
-                <Menu
-                    placement='bottom-end'
-                    trigger={({ getTriggerProps, triggerRef }) => (
-                        <Icon
-                            icon={'help-circle'}
-                            {...getTriggerProps({ innerRef: triggerRef })}
-                            isLink
-                            size={'25px'}
-                            color={colors.white}
-                        />
-                    )}
-                    triggerOn='click'
-                >
-                    <MenuDiv>{menuItems}</MenuDiv>
-                </Menu>
-            </HeaderMenuUserDiv>
-            {props.headerDetails && props.headerDetails.userId && (
+    }
+
+    render() {
+        let props = this.props;
+        let reviewNotificationRedDot = '';
+        let icon = 'notification';
+        let clickIcon = this.onClickFunction.bind(
+            this,
+            PageUtils.workflowAdminPage()
+        );
+        if (props.pending || props.reviewRoles || props.reviewGroups) {
```

Comment:
Will that be better to put all these conditions together? And if necessary, we can have a function to check the status of pending members, or review roles or groups, and you can just pass these values to that function

Reply:
ok good suggestion, i can definitely write a helper function to simplify this to 
```
if (shouldAlertUser(...))
```

Reply:
done.